mesonbuild/modules/pkgconfig.py

```diff
@@ -338,18 +360,17 @@ def _generate_pkgconfig_file(self, state, deps, subdirs, name, description,
         else:
             outdir = state.environment.scratch_dir
             prefix = PurePath(coredata.get_option(mesonlib.OptionKey('prefix')))
-        # These always return paths relative to prefix
-        libdir = PurePath(coredata.get_option(mesonlib.OptionKey('libdir')))
-        incdir = PurePath(coredata.get_option(mesonlib.OptionKey('includedir')))
         fname = os.path.join(outdir, pcfile)
         with open(fname, 'w', encoding='utf-8') as ofile:
-            if not dataonly:
-                ofile.write('prefix={}\n'.format(self._escape(prefix)))
-                if uninstalled:
-                    ofile.write('srcdir={}\n'.format(self._escape(srcdir)))
-                if deps.pub_libs or deps.priv_libs:
-                    ofile.write('libdir={}\n'.format(self._escape('${prefix}' / libdir)))
-                ofile.write('includedir={}\n'.format(self._escape('${prefix}' / incdir)))
+            for optname in optnames:
+                if optname in referenced_vars:
+                    if optname == 'prefix':
+                        ofile.write('prefix={}\n'.format(self._escape(prefix)))
+                    else:
+                        dirpath = PurePath(coredata.get_option(mesonlib.OptionKey(optname)))
+                        ofile.write('{}={}\n'.format(optname, self._escape('${prefix}' / dirpath)))
```

Comment:
I think it needs to be `dirpath.as_posix()`

Reply:
Nope, that's explicitly handled by `self._escape()` and the argument *to* `self._escape` needs to use the overloaded `/` to join itself.

Take a look at the old version of this code, it didn't use `.as_posix()` either.

Reply:
nevermind, was already like that so I gave it a try and it converts to posix already.